netcontrold/tests/lib/test_prelim.py

```diff
@@ -0,0 +1,111 @@
+#
+#  Copyright (c) 2020 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+from unittest import TestCase
+from netcontrold.lib import dataif
+from netcontrold.lib.error import NcdException, ObjConsistencyExc
+
+
+class TestDataif_add_rxq(TestCase):
+    """
+    Test for adding rxq to an existing port.
+    """
+
+    # setup test environment
+    def setUp(self):
+        # create one port object
+        self.port1 = dataif.Port("port1")
+
+    # add rxq to port object
+        rxq1 = self.port1.add_rxq(1)
+
+    # Test case:
+    #   adding rxq to an existing port and checking whether
+    #   it is added or not.
+    def test_port_add_rxq_1(self):
+        # add rxq to port object
+        rxq2 = self.port1.add_rxq(2)
+
+        self.assertEqual(self.port1.find_rxq_by_id(2), rxq2)
+
+    # Test case:
+    #   adding already existing rxq to port and checking whether
+    #   addition is skipped.
+    def test_port_add_rxq_2(self):
+        with self.assertRaises(ObjConsistencyExc):
```

Comment:
This is the actual test for existing port. Correct prev test then.

Reply:
Also, why "with" clause you try this.. i gave a one line to test in previous review.